src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1034,79 +1033,92 @@ std::string primitive_inst::get_implementation_name() const {
-        } else {
-            mem_lock<char, mem_lock_type::read> lock{_outputs[0], get_node().get_program().get_stream()};
-            ob << cldnn::make_data(lock.data(), data_size);
-        }
-    } else {
-        object_type _object_type = object_type::EXECUTABLE_INST;
-        ob << cldnn::make_data(&_object_type, sizeof(object_type));
-
-        kernel_arguments_data args = _impl->get_arguments(*this);
-        kernel_arguments_data_idx args_idx;
-        convert_args(args, args_idx);
-        _impl->set_arguments(args_idx);
-
-        _impl_params->save(ob);
-        ob.setKernlImplParams(_impl_params.get());
-        ob << _impl;
-
-        ob << _node_output_layout;
-        ob << has_mutable_input();
-        ob << mem_allocated();
-        ob << is_dynamic();
-        ob << _node->get_primitive()->type_string();
-        ob << id();
-        ob << org_id();
-        ob << is_input();
-        ob << is_output();
-        ob << inputs_memory_count();
-        ob << outputs_memory_count();
-        ob << get_fused_mem_count();
-        ob << get_fused_mem_offset();
-        ob << can_be_optimized();
-        ob << can_share_buffer();
-        ob << is_constant();
-
-        ob << _outputs[0]->get_layout();
-        const auto _allocation_type = _outputs[0]->get_allocation_type();
-        ob << make_data(&_allocation_type, sizeof(_allocation_type));
+    const network& const_network = p_inst->get_network();
+    while (head < queue.size()) {
+        auto curr_item = queue.at(head);
+        auto curr_prim = const_network.get_primitive(curr_item.first);
 
-        ob << _node->get_memory_dependencies();
+        if (p_inst->get_network().get_engine().is_the_same_buffer(mem_ptr, curr_prim->output_memory()))
+            return curr_prim->id();
 
-        ob << _deps.size();
-        for (const auto& dep : _deps) {
-            ob << dep->id();
-        }
+        if (max_dist > curr_item.second)
+            for (auto& p_inst : curr_prim->dependencies())
+                queue.emplace_back(std::make_pair(p_inst->id(), curr_item.second+1));
 
-        ob << _exec_deps.size();
-        for (const auto& dep : _exec_deps) {
-            ob << dep->id();
-        }
+        head += 1;
+    }
 
-        ob << _intermediates_memory.size();
-        for (const auto& ibuf : _intermediates_memory) {
-            ob << ibuf->get_layout();
-            const auto _allocation_type = ibuf->get_allocation_type();
-            ob << make_data(&_allocation_type, sizeof(_allocation_type));
-        }
+    return "NOT_FOUND";
+}
+
+// Cache blob format:
+//     [ primitive type ]
+//     [ kernel_arguments ]
+//     [ kernel_impl_params ]
+//     [ primitive_impl ]
+//     [ member variables of primitive_inst ]
+//     [ output memory information ]
+//     [ memory dependency information ]
+//     [ execution dependency information ]
+//     [ intermediate memory information ]
+void primitive_inst::save(cldnn::BinaryOutputBuffer& ob) const {
```

Comment:
https://github.com/openvinotoolkit/openvino/pull/13801#discussion_r1017999467
I overrided save and load methods for data/mutable data, and removed a branch. Thank you.